index.bs

```diff
@@ -2222,6 +2266,99 @@ To <dfn>parse trigger specs</dfn> given a [=map=] |map|, a [=moment=]
+1. If |sourceType| is "<code>[=source type/event=]</code>" and |states| is greater than [=default max number of event states=]
+    and |map|["`max_event_states`"] does not [=map/exist=], return an error.
+1. Let |maxNumEventStates| be the result of running
+    [=parse an optional 32-bit unsigned integer=] with |map|,
+    "`max_event_states`", and [=default max number of event states=].
+1. If |maxNumEventStates| is an error or |maxNumEventStates| is equal to 0 or |maxNumEventStates| is greater than [=max trigger-state cardinality=], return an error.
+1. If |sourceType| is "<code>[=source type/event=]</code>" and |states| is greater than |maxNumEventStates|, return an error.
+1. Return |maxNumEventStates|.
+
+To <dfn>parse attribution scopes</dfn> from a [=map=] |map| and a possibly null 32-bit unsigned integer |attributionScopeLimit|:
+1. If |map|["`attribution_scopes`"] does not [=map/exist=], return null.
+1. Let |result| be a new [=set=].
+1. If |map|["`attribution_scopes`"] [=map/exists=]:
+    1. Let |values| be |map|["`attribution_scopes`"].
+    1. If |values| is not a [=list=], return an error.
+    1. [=list/iterate|For each=] |value| of |values|:
+        1. If |value| is not a [=string=], return an error.
+        1. If |value|'s [=string/length=] is greater than [=max length per attribution scope=], return an error.
+        1. [=set/Append=] |value| to |result|.
+1. If |result| is not [=set/is empty|empty=]:
+    1. If |attributionScopeLimit| is null, return an error.
+1. If |result|'s [=set/size=] is greater than |attributionScopeLimit|, return an error.
+1. If |result|'s [=set/size=] is greater than [=max number of attribution scopes per source=], return an error.
+1. Return |result|.
+
+To <dfn>find sources with common destinations and reporting origin</dfn> given an [=attribution source=] |pendingSource|:
+1. Let |cache| be the user agent's [=attribution source cache=].
+1. Let |matchingSources| be a new [=list=].
+1. [=set/iterate|For each=] |source| of |cache|:
+    1. Let |commonDestinations| be the [=set/intersection=] of |source|'s [=attribution source/attribution destinations=] and |pendingSource|'s [=attribution source/attribution destinations=].
+    1. If |commonDestinations| [=set/is empty=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/reporting origin=] and |pendingSource|'s [=attribution source/reporting origin=] are not [=same origin=], [=iteration/continue=].
+    1. [=list/Append=] |source| to |matchingSources|.
+1. Return |matchingSources|.
+
+To <dfn>Remove event reports for source</dfn> given an [=attribution source=] |source| and a [=moment=] |minTriggerTime|:
+1. Let |matchingReports| be a new [=list=] whose elements are all the elements in the [=event-level report cache=] whose [=event-level report/trigger time=] is greater than or equal to |minTriggerTime| and [=event-level report/event Id=] and |source|'s [=attribution source/event ID=].
```

Comment:
IIUC this only applies to fake reports. Currently for fake reports the trigger time is set to the greatest moment of the report window (https://wicg.github.io/attribution-reporting-api/#obtain-a-fake-report). Do we have to delete the fake reports if the start of the report window is earlier than the current time? 
cc @vikassahu29

Reply:
This is to hide the fact the deactivated source was noised. Tagged you in the internal docs for this section.

Reply:
updated the `triggerTime` logic for fake report.